src/components/map/Layers/Countries.tsx

```diff
@@ -1,133 +1,156 @@
-import { EstimateGradePrevalence } from "types";
+import { EstimateGradePrevalence, CountriesMapConfig } from "types";
 import mapboxgl from "mapbox-gl";
 import CountryPopup from 'components/map/Popups/CountryPopup'
+import PartnershipsConfig from '../../../PartnershipsConfig'
+import { useHistory } from 'react-router-dom';
+
+const COUNTRY_LAYER_ID = 'Countries';
 
 // Maps estimate grade prevalence data to a match ISO3 code in the countries feature layer
 function SetCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
+
+    map.removeFeatureState({
+        source: COUNTRY_LAYER_ID,
+        sourceLayer: COUNTRY_LAYER_ID
+        });
+
     estimateGradePrevalences.forEach((country: EstimateGradePrevalence) => {
         if (country && country.testsAdministered && country.alpha3Code) {
+
+            const partnershipConfig = PartnershipsConfig.find(x => x.iso3 === country.alpha3Code)
+
             map.setFeatureState(
                 {
-                    source: "Countries",
-                    sourceLayer: "Countries",
+                    source: COUNTRY_LAYER_ID,
+                    sourceLayer: COUNTRY_LAYER_ID,
                     id: country.alpha3Code,
                 },
                 {
                     hasData: true,
+                    isHighlighted: false,
                     testsAdministered: country.testsAdministered,
                     geographicalName: country.geographicalName,
                     numberOfStudies: country.numberOfStudies,
                     localEstimate: country.localEstimate,
                     nationalEstimate: country.nationalEstimate,
                     regionalEstimate: country.regionalEstimate,
                     sublocalEstimate: country.sublocalEstimate,
+                    partnershipConfig: partnershipConfig
                 }
             );
         }
     });
 }
 
-// Filters out Country features based on their existance within the new estimate grade prevalences
-function FilterCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
-    const onlyIso3: (string | null)[] = estimateGradePrevalences.map((c) => { return c.alpha3Code })
-
-    if (map.getLayer("Countries") && map.isSourceLoaded('Countries')) {
-        map?.setFilter('Countries',
-            ["in",
-                ["get", 'CODE'],
-                ["literal", onlyIso3]
-            ]);
-    }
-    else { // If countries has not been loaded, wait until it's source has finished.
-        map.on('sourcedata', function (e: any) {
-            if (e.sourceId === 'Countries' && map.isSourceLoaded('Countries')) {
-                map?.setFilter('Countries',
-                    ["in",
-                        ["get", 'CODE'],
-                        ["literal", onlyIso3]
-                    ]);
-            }
+function SetStudiesMode(map: mapboxgl.Map, iso3Name: string) {
+    map.removeFeatureState({
```

Comment:
So it looks like we are completely wiping the map state here. I think this should work for the current use case (we only use map state to track isHighlighted, and only one country should be highlighted at a time), however if we end up using state to track other properties in the future this implementation wouldn't work as those would get wiped.

Can we add a comment describing this shortcoming to our code for future reference?

Reply:
yep, or I can just remove map highlight instead of doing a wipe. I'll look into how easy it would be to change

Reply:
Just left a comment for now.